resources/assets/lib/user-multiplayer-index/room.tsx

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+import DifficultyBadge from 'difficulty-badge';
+import Img2x from 'img2x';
+import RoomJson from 'interfaces/room-json';
+import { maxBy, minBy } from 'lodash';
+import { computed } from 'mobx';
+import { observer } from 'mobx-react';
+import * as moment from 'moment';
+import * as React from 'react';
+import { StringWithComponent } from 'string-with-component';
+import { UserLink } from 'user-link';
+import UserMultiplayerHistoryContext from 'user-multiplayer-history-context';
+import { getDiffRating } from 'utils/beatmap-helper';
+import { classWithModifiers } from 'utils/css';
+
+interface Props {
+  room: RoomJson & Required<Pick<RoomJson, 'playlist'>>;
+}
+
+const endingSoonDiffMs = 60 * 60 * 1000; // 60 minutes.
+
+@observer
+export default class Room extends React.Component<Props> {
+  static contextType = UserMultiplayerHistoryContext;
+  declare context: React.ContextType<typeof UserMultiplayerHistoryContext>;
+
+  @computed
+  get status() {
+    if (!this.props.room.active) {
+      return 'ended';
+    }
+
+    const diff = new Date(this.props.room.ends_at).getTime() - new Date().getTime();
+
+    return diff < endingSoonDiffMs ? 'soon' : 'active';
+  }
+
+  @computed
+  get maxDifficulty() {
+    const max = maxBy(this.props.room.playlist, (playlist) => this.context.beatmaps.get(playlist.beatmap_id)?.difficulty_rating);
+    return this.context.beatmaps.get(max?.beatmap_id ?? 0)?.difficulty_rating ?? 0;
+  }
+
+  @computed
+  get minDifficulty() {
+    const min = minBy(this.props.room.playlist, (playlist) => this.context.beatmaps.get(playlist.beatmap_id)?.difficulty_rating);
+    return this.context.beatmaps.get(min?.beatmap_id ?? 0)?.difficulty_rating ?? 0;
+  }
+
+  get background() {
+    const beatmap = this.context.beatmaps.get(this.props.room.playlist[0].beatmap_id);
+    const beatmapset = this.context.beatmapsets.get(beatmap?.beatmapset_id ?? 0);
+
+    if (beatmapset == null) return undefined;
+
+    return beatmapset.covers.cover;
+  }
+
+  render() {
+    const endsAt = moment(this.props.room.ends_at);
+
+    return (
+      <div className='multiplayer-room'>
```

Comment:
apparently I forgot about it but a link to the room detail could be useful? (`/multiplayer/rooms/:room`)

Reply:
I thought it's supposed to go to lazer instead 🤔

Reply:
well, any link is better than nothing (and can be updated later as needed)

Reply:
Won't that endpoint not work due to the rooms being in a completely different table structure?

Reply:
that's the new multiplayer room leaderboard endpoint

Reply:
Damn, totally forgot this was a thing!